src/transformers/tokenization_marian.py

```diff
@@ -200,16 +200,19 @@ def save_vocabulary(self, save_directory: str) -> Tuple[str]:
         Returns:
             :obj:`Tuple(str)`: Paths to the files saved.
         """
-        save_dir = Path(save_directory)
-        assert save_dir.is_dir(), f"{save_directory} should be a directory"
-        save_json(self.encoder, save_dir / self.vocab_files_names["vocab"])
+        dest_dir = Path(save_directory)
+        assert dest_dir.is_dir(), f"{save_directory} should be a directory"
+        save_json(self.encoder, dest_dir / self.vocab_files_names["vocab"])
+        src_dir = Path(self.spm_files[0]).parent
 
         for orig, f in zip(["source.spm", "target.spm"], self.spm_files):
-            dest_path = save_dir / Path(f).name
+            dest_path = dest_dir / orig
             if not dest_path.exists():
-                copyfile(f, save_dir / orig)
-
-        return tuple(save_dir / f for f in self.vocab_files_names)
+                copyfile(f, dest_path)
+        cfg_path = src_dir.joinpath("tokenizer_config.json")
+        if cfg_path.exists():
+            copyfile(cfg_path, "tokenizer_config.json")
+        return tuple(dest_dir / f for f in self.vocab_files_names)
```

Comment:
Are you handling the saving of `tokenizer_config.json` in the `save_vocabulary`? This should already be handled by the `save_pretrained` method in the `PreTrainedTokenizerBase`

Reply:
ah makes sense, will fix!